### ISSUE TITLE
Bug in g_path_regularize loss in multi-gpu case

### ISSUE THREAD

**Issue by rakhimovv** [rosinality/stylegan2-pytorch#159]
In https://github.com/rosinality/stylegan2-pytorch/blob/master/train.py#L96

there are lines
```
path_lengths = torch.sqrt(grad.pow(2).sum(2).mean(1))
path_mean = mean_path_length + decay * (path_lengths.mean() - mean_path_length)
path_penalty = (path_lengths - path_mean).pow(2).mean()
```
However this would not produce the same penalty in multi-gpu case compared to single-gpu case if we consider the same total batch size.

Instead there should be something like:
```
path_lengths = torch.sqrt(grad.pow(2).sum(2).mean(1))
path_mean_cur = all_reduce_mean(path_lengths)
path_mean = mean_path_length + decay * (path_mean_cur - mean_path_length)
path_penalty = (path_lengths - path_mean).pow(2).mean()
```

As an analogy one can think about `nn.SyncBatchNorm` vs `nn.BatchNorm`

**Comment by rosinality** [comment]
Yes, but I don't know synchronization is needed for training. (I think official repository does not sync statistics.)
